src/services/AuthService.ts

```diff
@@ -15,11 +19,62 @@ export class AuthService {
     this.transactions = new TransactionsManager(entityManager);
   }
 
-  public async createUser(user: CreateUserRequest): Promise<PrivateProfile> {
+  public async loginUser(authRequest: AuthRequest): Promise<UserSessionModel | undefined> {
+    // checks if the email is a Cornell email
+    const emailIndex = authRequest.user.email.indexOf('@cornell.edu')
+    if (emailIndex === -1) {
+      throw new Error('Non-Cornell email used!');
+    } 
+
+    if (process.env.OAUTH_ANDROID_CLIENT && process.env.OAUTH_IOS_ID) {
+      // verifies info using id token
+      const ticket = await client.verifyIdToken({
+        idToken: authRequest.idToken,
+      });
+    
+      const payload = ticket.getPayload();
+
+      if (payload){
+        // token checking in payload
+        if (process.env.IS_PROD?.toLowerCase() === "true" &&
```

Comment:
why are we calling this function if `IS_PROD = false` in the first place?

Reply:
I want ppl working on it to be able to test google login (i.e. `IS_PROD = false`). If `IS_PROD = true`, then I check if the request is being sent from an iOS or Android client.